packages/core/src/browser/drag-and-drop.ts

```diff
@@ -0,0 +1,137 @@
+/*
+* Copyright (C) 2017 Ericsson and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+import { injectable, inject } from "inversify";
+import { TreeWidget, TreeProps, ITreeModel, ITreeNode, NodeProps } from './tree';
+import { ContextMenuRenderer } from "./context-menu-renderer";
+import { ElementAttrs } from "@phosphor/virtualdom";
+
+export const DROPZONE_CLASS = 'dropzone';
+
+@injectable()
+export class DragAndDraggable extends TreeWidget {
```

Comment:
OK, in the next version, the "drag-and-drop" class code will be available inside the "FileTreeWidget" class, no new class

Reply:
Note we may need a way to disable that so that it's not availabe when doing file->open for example.

@akosyakov  ideas on how to differentiate the two?

Reply:
It should be an optional feature, configured with `FileTreeProps`.